operator/src/test/java/org/bf2/operator/operands/KafkaClusterTest.java

```diff
@@ -276,6 +277,25 @@ void testManagedKafkaToKafkaWithSizeChanges() throws IOException {
         diffToExpected(larger, "/expected/strimzi.yml", "[{\"op\":\"replace\",\"path\":\"/spec/kafka/config/client.quota.callback.static.storage.soft\",\"value\":\"28633115306\"},{\"op\":\"replace\",\"path\":\"/spec/kafka/config/client.quota.callback.static.storage.hard\",\"value\":\"28633115306\"},{\"op\":\"replace\",\"path\":\"/spec/kafka/storage/volumes/0/size\",\"value\":\"41383100373\"}]");
     }
 
+    @Test
+    void numRecoveryThreadShouldConsiderStorageCount() throws IOException {
+        //Given
+        final int volumeCount = 2;
+        alternativeConfig(clone -> clone.getKafka().setVolumeCount(volumeCount));
+        final ManagedKafka managedKafka = dummyManagedKafka("0");
+
+        //When
+        Kafka kafka = kafkaCluster.kafkaFrom(managedKafka, null);
+
+        //Then
+        final String actualRecoveryThreadNum = String.valueOf(kafka.getSpec().getKafka().getConfig().get("num.recovery.threads.per.data.dir"));
+
+        Quantity cpu = new Quantity(configs.getConfig(managedKafka).getKafka().getContainerCpu());
+        BigDecimal cpuBytes = Quantity.getAmountInBytes(cpu);
+        int cpuCores = Math.max(1, cpuBytes.intValue());
+        assertEquals(String.valueOf(cpuCores/volumeCount), actualRecoveryThreadNum);
```

Comment:
This effectively reproduces the production code so to me it's not a valid test.

Personally I'd hard code the test expectations to ensure the production code generates the correct results.

Reply:
Good point! Thanks for the suggestion!